pkg/eestream/crc_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package eestream
+
+import (
+	"encoding/binary"
+	"hash/crc32"
+
+	"storj.io/storj/pkg/eestream/ranger"
+)
+
+type crcAdder struct {
+	Table *crc32.Table
+}
+
+func newCRCAdder(t *crc32.Table) *crcAdder {
+	return &crcAdder{Table: t}
+}
+
+func (c *crcAdder) InBlockSize() int  { return 64 }
+func (c *crcAdder) OutBlockSize() int { return 64 + 4 + 8 }
+
+func (c *crcAdder) Transform(out, in []byte, blockOffset int64) (
+	[]byte, error) {
+	out = append(out, in...)
+	var buf [8]byte
+	binary.BigEndian.PutUint64(buf[:], uint64(blockOffset))
+	out = append(out, buf[:]...)
+	binary.BigEndian.PutUint32(buf[:4], crc32.Checksum(out, c.Table))
+	out = append(out, buf[:4]...)
```

Comment:
What, so special with the first half of the buffer? It would be really helpful to have a description of what this transform function exactly does.

Reply:
will do